src/Framework/PaymentGateways/DataTransferObjects/GatewayRouteData.php

```diff
@@ -39,9 +44,18 @@ public static function fromRequest(array $request)
         $self->gatewayId = $request['give-gateway-id'];
         $self->gatewayMethod = $request['give-gateway-method'];
         $self->routeSignature = isset($request['give-route-signature']) ? $request['give-route-signature'] : null;
+        $self->routeSignatureId = isset($request['give-route-signature-id']) ? $request['give-route-signature-id'] : null;
 
         $self->queryParams = array_filter($request, static function ($param) {
-            return !in_array($param, ['give-listener', 'give-gateway-id', 'give-gateway-method', 'give-route-signature']
+            return !in_array(
```

Comment:
Might be simpler with [array_intersect_key](https://www.php.net/manual/en/function.array-intersect-key.php).